addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -1433,14 +1434,22 @@ export class OdooEditor extends EventTarget {
         let start = range.startContainer;
         let end = range.endContainer;
         // Let the DOM split and delete the range.
+        const startBlock = closestBlock(start);
+        const endBlock = closestBlock(end);
+        const startLi = closestElement(start, 'li');
         const doJoin =
-            (closestBlock(start) !== closestBlock(range.commonAncestorContainer) ||
-            closestBlock(end) !== closestBlock(range.commonAncestorContainer))
-            && (closestBlock(start).tagName !== 'TD' && closestBlock(end).tagName !== 'TD');
+            (startBlock !== closestBlock(range.commonAncestorContainer) ||
+                endBlock !== closestBlock(range.commonAncestorContainer))
+            && (startBlock.tagName !== 'TD' && endBlock.tagName !== 'TD')
+            && (firstLeaf(startBlock) !== start || lastLeaf(endBlock) !== end);
         let next = nextLeaf(end, this.editable);
         const splitEndTd = closestElement(end, 'td') && end.nextSibling;
         const contents = range.extractContents();
         setSelection(start, nodeSize(start));
+        if (startLi && startLi.classList.contains('o_checked') &&
+            startLi.textContent === '' && closestElement(end, 'li') !== startLi) {
```

Comment:
what part are we handling here? can you write a scenario.

Reply:
This condition handles the part where we uncheck the checked list meaning for checklist like

- [x] [abc
- [ ] def]

if we delete this, the first checkbox should be unchecked, resulting in

- [ ] []

and `closestElement(end, 'li') !== startLi` for the case of

- [x] [abc]

this should result in 
- [x] []

Reply:
Please add a comment to make that clear.